pkg/pillar/cmd/zedagent/handleconfig.go

```diff
@@ -281,7 +279,60 @@ func getLatestConfig(url string, iteration int,
 	return inhaleDeviceConfig(config, getconfigCtx, false)
 }
 
-func validateConfigMessage(url string, r *http.Response) error {
+func getCloudCertChain(ctx *zedagentContext) bool {
+	// get Certs is always V2 API, if the reply turns out it's not, will reset it, use http for now
+	zedcloudCtx.V2API = true
+	certURL := zedcloud.URLPathString(serverNameAndPort, zedcloudCtx.V2API, true, nilUUID, "certs")
+	resp, contents, rtf, err := zedcloud.SendOnAllIntf(zedcloudCtx, certURL, 0, nil, 0, false)
+	if err != nil {
+		if rtf == types.ExtStatusRemTempFail {
+			log.Infof("getCloudCertChain remoteTemporaryFailure: %s", err)
+		} else {
+			log.Errorf("getCloudCertChain failed: %s", err)
+		}
+		return false
+	}
+
+	switch resp.StatusCode {
+	case http.StatusOK, http.StatusCreated, http.StatusNotModified:
+		log.Infof("getCloudCertChain: status %s\n", resp.Status)
+	case http.StatusNotFound, http.StatusUnauthorized, http.StatusNotImplemented, http.StatusBadRequest:
+		log.Infof("getCloudCertChain: server %s does not support V2 API\n", serverName)
+		zedcloudCtx.V2API = false
+		return false
+	default:
+		log.Errorf("getCloudCertChain: statuscode %d %s", resp.StatusCode, http.StatusText(resp.StatusCode))
+		return false
+	}
+
+	err = validateProtoMessage(certURL, resp)
+	if err != nil {
+		log.Errorf("getCloudCertChain: resp header error\n")
+		return false
+	}
+
+	err = zedcloud.VerifyCloudCertChain(zedcloudCtx, serverName, contents)
+	if err != nil {
+		log.Infof("getCloudCertChain: verify err %v\n", err)
+		return false
+	}
+	log.Infof("getCloudCertChain: success\n")
+	return true
+}
+
+func triggerFetchCerts(ctx *zedagentContext) {
+	// trigger a one sec timer to acquire new certs from cloud
+	interval := time.Duration(1)
```

Comment:
I believe that is 1 nanosecond and not one second.

Reply:
got it.